### PR TITLE
Add sha3_256 opcode to runtime

### PR DESCRIPTION
**Pull request by thdailong**
Closes #x

## Proposed Changes

- Add sha3_256 opcode to runtime

## TODO

- [x] update change log

### CODE REVIEW

File: packages/runtime/src/interpreter/opcode-list.ts [hunk 1]
```diff
@@ -5046,4 +5046,40 @@ export class Replace2 extends Replace {
...
 
+}
+
+
+// SHA3_256 hash of value A, yields [32]byte
+// https://github.com/phusion/node-sha3#generating-a-sha3-512-hash
+// push to stack [...stack, bytes]
+export class Sha3_256 extends Op {
+	readonly line: number;
+	readonly interpreter: Interpreter;
```
Comment (vuvoth) [inline packages/runtime/src/interpreter/opcode-list.ts:5057]: I don't this this opcode required interpreter.

**Review by vuvoth — COMMENTED** [review]
Thanks for your PR! I left a few comment.

File: packages/runtime/test/src/interpreter/opcode-list.ts [hunk 2]
```diff
@@ -7138,4 +7139,51 @@ describe("Teal Opcodes", function () {
...
+				"ae39517df229f45df862c060e693c0e69691dac70fa65605a62fabad8029a4e7",
+				"hex"
+			);
+
+			const top = stack.pop();
+			assert.deepEqual(expected, top);
+		});
+
+		it(
+			"should throw invalid type error Keccak256",
```
Comment (robert-zaremba) [inline packages/runtime/test/src/interpreter/opcode-list.ts:7162]: Please improve the error description. Why the invalid type error is thrown?

**Comment by robert-zaremba** [comment]
I've added a commit to introduce an abstract class for hash operations (to have less copy-paste code).

**Review by robert-zaremba — APPROVED** [review]

**Pull request merged**